### PR TITLE
Add ability to add unix socket sql dsn

### PR DESCRIPTION
**Pull request by michaelliu2**

### CODE REVIEW

**Comment by michaelliu2** [comment]
@ianlofs @hdahme @katekligman 
Ability to use unix socket dsn for sql databases (so we can use cloudsql)

Format of unix socket dsn : mysql://user:password@unix(socket_path)/database

File: etl_framework/datastores/mysql_database.py [hunk 1]
```diff
@@ -18,9 +18,23 @@ def _create_connection(self):
         returns new sql connection object
         """
 
-        return MySQLdb.connect(host=self.credentials[0],
-                              port=self.credentials[1],
-                              user=self.credentials[2],
-                              passwd=self.credentials[3],
-                              db=self.credentials[4],
-                              charset='utf8')
+        if len(self.credentials) == 4:
```
Comment (hdahme) [inline etl_framework/datastores/mysql_database.py]: Comments!

File: etl_framework/datastore_interfaces/sql_database_interface.py [hunk 2]
```diff
@@ -46,20 +46,34 @@ def parse_dsn(dsn):
...
+
+            return (host, port, user, password, database)
 
     def set_credentials(self, credentials):
         """sets the database credentials"""
 
-        if len(credentials) != 5:
-            raise Exception('Db credentials must be array of 5 items. %d given'%len(credentials))
+        if len(credentials) not in [4, 5]:
+            raise Exception('Db credentials must be array of 3 or 5 items. %d given'%len(credentials))
```
Comment (hdahme) [inline etl_framework/datastore_interfaces/sql_database_interface.py]: Maybe also explain what the credentials must be in the exception

**Comment by hdahme** [comment]
Otherwise, 👍 ⬛ 🐎

**Pull request merged**